### PR TITLE
Support schema validation test with mocha. 

### PR DESCRIPTION
**Pull request by Tiano2017**

### CODE REVIEW

File: tools/runSchemaValidation.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	let textDocument = Types.TextDocument.create(filePath, 'json', 0, content);
+	let jsonDocument = service.parseJSONDocument(textDocument);
+
+	return service.doValidation(textDocument, jsonDocument);
+};
+
+describe('validate template files with schema - ', () => {
+	let files = fs.readdirSync(testFolder);
+
+	// // it.each(files, "running schema validation on '%s'", ['element'], function (element) {
```
Comment (vivsriaus) [inline tools/runSchemaValidation.js]: :)

File: tools/runSchemaValidation.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	// // 	// console.log(validateResult);
+	// // });
+
+	forEach(files).it("running schema validation on '%s'", function (file, done) {
+		this.timeout(5000);
+
+		validateTemplateWithSchema(file)
+			.then(function (result) {
+					assert.deepEqual(result, [], 'there are validation errors.');
```
Comment (vivsriaus) [inline tools/runSchemaValidation.js]: may be include the file name here in the message?

File: tools/runSchemaValidation.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	// // 	// 	testDone(error);
+	// // 	// })
+	// // 	//.catch((err) => { console.log(err); })
+	// // 	//.then(() => next());
+
+	// // 	// console.log(validateResult);
+	// // });
+
+	forEach(files).it("running schema validation on '%s'", function (file, done) {
+		this.timeout(5000);
```
Comment (vivsriaus) [inline tools/runSchemaValidation.js]: why do we need the timeout?

File: tools/runSchemaValidation.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	}
+};
+
+let parameters = {
+	schemaRequestService: requestServiceMock,
+	workspaceContext: workspaceContext
+};
+
+let service = SchemaService.getLanguageService(parameters);
+let testFolder = './templateFiles';
```
Comment (vivsriaus) [inline tools/runSchemaValidation.js]: rename testFolder to a better descriptive name, templatesFolder?

File: tools/runSchemaValidation.js (hunk 3 above)
Comment (Tiano2017) [reply tools/runSchemaValidation.js]: The default timeout is 2 seconds which is not enough here.

File: tools/runSchemaValidation.js (hunk 4 above)
Comment (Tiano2017) [reply tools/runSchemaValidation.js]: sure. renamed.

File: tools/runSchemaValidation.js (hunk 1 above)
Comment (Tiano2017) [reply tools/runSchemaValidation.js]: will remove the comments.

File: tools/runSchemaValidation.js (hunk 2 above)
Comment (Tiano2017) [reply tools/runSchemaValidation.js]: fixed.

**Comment by Tiano2017** [comment]
@vivsriaus  yes, I'm going to run this manually for a while and then enable it with Travis.

**Pull request merged**